sys/kern/src/arch/rv64/mtimer.rs

```diff
@@ -28,17 +30,19 @@ const MTIME: u64 = 0x0200_BFF8;
 // back to 0 on each interrupt.
 //
 #[no_mangle]
-pub unsafe fn set_timer(tick_divisor: u32) {
+pub unsafe fn set_timer() {
     // Set high-order bits of mtime to zero.  We only call this function prior
     // to enabling interrupts so it should be safe.
     unsafe {
-        core::ptr::write_volatile(MTIME as *mut u64, 0);
-        core::ptr::write_volatile(MTIMECMP as *mut u64, tick_divisor.into());
+        let time = core::ptr::read(MTIME as *const u64);
```

Comment:
This change is not related to the rest of the series.

Reply:
Practically, some part of this change had to be included (the signature change, and removing the mtimecmp set on the now-removed argument. It could probably have been made as a smaller change - setting mtimecmp to 0 to match, for example, but the delta is small and impactful enough that just moving it to the canonical behaviour while adjusting the signature seemed like it would produce a clearer change history. The alternative being a commit that changes it to something counterintuitive, and then to the canonical flow

Reply:
Since this PR is dealing with S-mode support, I think it would have been cleaner to separate out the API change to timer and hence M-mode outside this PR. Something like:

1. PR to address the issue for M-mode timer where mtime is updated instead of mtimecmp. This would address SW-605. It would include the API update which explains why it is being changed.
2. PR to add S-mode support using the new API.

I understand it is small enough that we can push it in, but if for instance the small change contains a bug, whoever ends up debugging this will have to look at a pretty large change rather than a few lines to bisect what went wrong. At this point, we don't really have a lot of users/developers, so I think its fine to keep this change as is. But, long term, we might want to create PRs using the smallest logical unit that makes sense.